lib/src/components/NxBreadcrumb/NxBreadcrumb.tsx

```diff
@@ -0,0 +1,99 @@
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import classnames from 'classnames';
+import { append, dropLast, head, init, insert, last, map, tail, takeLast } from 'ramda';
+import { faEllipsisH } from '@fortawesome/free-solid-svg-icons';
+
+import NxTextLink from '../NxTextLink/NxTextLink';
+import { Props, Crumb, DropdownProps, propTypes } from './types';
+import NxIconDropdown from '../NxIconDropdown/NxIconDropdown';
+
+import './NxBreadcrumb.scss';
+import NxOverflowTooltip from '../NxTooltip/NxOverflowTooltip';
+
+export { Props, Crumb };
+
+function CurrentBreadcrumb({ name }: Crumb) {
+  return (
+    <NxOverflowTooltip>
+      <li className="nx-breadcrumb__list-item nx-breadcrumb__list-item--link-container">
+        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+        <a className="nx-breadcrumb__link nx-breadcrumb__link--current" aria-current="page">{name}</a>
+      </li>
+    </NxOverflowTooltip>
+  );
+}
+
+function BreadcrumbLink({ name, href }: Crumb) {
+  return (
+    <NxOverflowTooltip>
+      <li className="nx-breadcrumb__list-item nx-breadcrumb__list-item--link-container">
+        <NxTextLink className="nx-breadcrumb__link" href={href}>{name}</NxTextLink>
+      </li>
+    </NxOverflowTooltip>
+  );
+}
+
+function BreadcrumbDropdownLink({ name, href }: Crumb) {
+  return (
+    <NxOverflowTooltip>
+      <NxTextLink className="nx-dropdown-link" href={href}>{name}</NxTextLink>
+    </NxOverflowTooltip>
+  );
+}
+
+function BreadcrumbDropdown({ crumbs, isOpen, onToggleCollapse }: DropdownProps) {
+  return (
+    <li className="nx-breadcrumb__list-item">
+      <NxIconDropdown isOpen={isOpen} onToggleCollapse={onToggleCollapse} icon={faEllipsisH} title="more…">
+        {map(c => <BreadcrumbDropdownLink key={c.href} { ...c } />, crumbs)}
+      </NxIconDropdown>
+    </li>
+  );
+}
+
+export default function NxBreadcrumb(props: Props) {
+  const { crumbs, isDropdownOpen, onToggleDropdown, className: classNameProp, ...otherProps } = props,
+      crumbBeforeDropdown = crumbs.length < 2 ? null : head(crumbs);
+
+  if (crumbBeforeDropdown) {
+    const dropdownCrumbs = crumbs.length < 5 ? [] : tail(dropLast(2, crumbs)),
+        crumbsAfterDropdown = crumbs.length < 5 ? tail(crumbs) : takeLast(2, crumbs),
+        crumbsToBecomeLinks = [crumbBeforeDropdown, ...init(crumbsAfterDropdown)],
+
+        // if we're in this block, the list was at least two items long so last on the "after" crumbs will
+        // always have a value
+        /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+        currentCrumb = last(crumbsAfterDropdown)!,
+        currentCrumbRender = <CurrentBreadcrumb key={currentCrumb.href} { ...currentCrumb } />,
+        rowLinks = map(c => <BreadcrumbLink key={c.href} { ...c } />, crumbsToBecomeLinks),
+        dropdown = dropdownCrumbs.length ?
+          <BreadcrumbDropdown key="%%nx-breadcrumb-dropdown"
```

Comment:
Is the "%%" here on purpose?

Reply:
> Is the "%%" here on purpose?

Yes; all of the other items in this list use the `href` values (which should be URLs) as their keys.  Keys need to be unique of course, so the way to ensure that this one is unique is to make it a string that cannot be a valid `href`.  This was the first way I thought of to do so - `%` is the escape character in URLs and should always (in URLs) be followed by two hex digits, so following it with another `%` instead is one way to construct a string that is not a valid URL or URL part.

Reply:
Thanks for the explanation.